### PR TITLE
Config reloading now detects changes to event listeners.  Fixes #339

### LINKED ISSUE #339: reread and update ignore changes to event listeners
If I change the configuration file for an event listener:

```
$ nano /etc/supervisor/conf.d/event_listener.conf
$ supervisorctl reread
No config updates to processes
$ supervisorctl update
$
```

If I change the configuration file for a program:

```
$ nano /etc/supervisor/conf.d/program.conf
$  sudo supervisorctl reread                    
riemann-dash: changed
$ sudo supervisorctl update                         
riemann-dash: stopped
riemann-dash: updated process group
```

### PR DESCRIPTION
**Pull request by maihde**
As reported in issue #339, changes to an event listener configuration don't get noticed during an "update" call.  This pull request includes a unittest to demostrate the problem along with a fix that causes the unittest to pass.

Note, this fix only ensures that the changes trigger as part of diff_to_activate.  I have not check to see if there are any side-effects associated with doing this for eventlisteners.

### CODE REVIEW

**Comment by mnaberez** [comment]
Thanks for this patch.  I'll test this out and will try to get it in the next point release if there aren't any issues.

**Comment by rparkhunovsky** [comment]
Any update? Fairly necessary feature!
Thanks for the PR!

File: supervisor/options.py [hunk 1]
```diff
@@ -1950,7 +1950,11 @@ def __eq__(self, other):
         if not isinstance(other, EventListenerPoolConfig):
             return False
 
-        if (self.name == other.name) and (self.priority == other.priority):
+        if ((self.name == other.name) and
+            (self.priority == other.priority) and
+            (self.process_configs == other.process_configs) and
+            (self.pool_events == other.pool_events) and
```
Comment (mnaberez) [inline supervisor/options.py]: `buffer_size` is missing.

**Comment by razvanphp** [comment]
@maihde can you please fix this as per @mnaberez suggestions?

Maybe we can get this merged and released soon, very frustrating.

Thank you!

**Comment by maihde** [comment]
Yep...I will get that changed tonight.

Cheers,
Michael

**Comment by akarl818** [comment]
Any updates on this?

**Pull request merged**